ofborg/src/tasks/build.rs

```diff
@@ -382,11 +392,59 @@ impl notifyworker::SimpleNotifyWorker for BuildWorker {
             .last();
         info!("----->8-----");
 
-        actions.build_finished(status, can_build, cannot_build_attrs);
+        let cachix_result = match &self.cachix {
+            Some(conf) => cache_build(conf.clone(), status.clone(), actions.log_snippet()),
+            None => {
+                info!("Cachix not configured, skipping");
+                false
+            }
+        };
+
+        actions.build_finished(status, can_build, cannot_build_attrs, cachix_result);
         info!("Done!");
     }
 }
 
+fn cache_build(cfg: CachixConfig, status: BuildStatus, log: Vec<String>) -> bool {
```

Comment:
Maybe `cache_push` would be a more appropriate name here. It doesn't look like we're building from the cache (though we are building *a* cache).

```suggestion
fn cache_push(cfg: &CachixConfig, status: BuildStatus, log: Vec<String>) -> bool {
```

My version of this function: https://gist.github.com/cole-h/00996da012d52ce8c95efb34743daf32

Reply:
It was meant as a verb like `cache that build` but I like `cache_push` better.